module_generator/README.md

```diff
@@ -0,0 +1,220 @@
+# Module Generator Template
+
+This is the Python module template for [CLI module generator](https://docs.viam.com/cli/#module).
+
+## How to Use
+
+1. Install cookiecutter: `pipx install cookiecutter`. If pipx is not an option, use `python -m pip install --user cookiecutter`.
```

Comment:
It isn't necessarily required to install cookiecutter with `pipx` for a one-off command by using `pipx run cookiecutter` instead: https://github.com/pypa/pipx?tab=readme-ov-file#walkthrough-running-an-application-in-a-temporary-virtual-environment

> For example, the blog post [How to set up a perfect Python project](https://sourcery.ai/blog/python-best-practices/) uses pipx run to kickstart a new project with [cookiecutter](https://github.com/cookiecutter/cookiecutter), a tool that creates projects from project templates.

This can be accomplished in one command without installing the viam-python-sdk:

```console
pipx run cookiecutter gh:viamrobotics/viam-python-sdk --directory module_generator/cookiecutter-template
```

Reply:
Very cool know that running it in one line is possible, but I am hitting an error when I try this command.

That being said, I don't think we'll be running the template with cookiecutter in the first place. We'll actually be calling it through the CLI, but this is just a beginning to set up the template.

Reply:
Who is this README for if this will be called through the CLI? 

>  since this PR isn't merged yet, the template isn't available through github
`cookiecutter` accepts a `checkout` flag to get templates from alternative branches in a git repo. I actually tested this template by running:

```console
pipx run cookiecutter gh:purplenicole730/viam-python-sdk --checkout RSDK-8553-create-python-module-template --directory module_generator/cookiecutter-template
```

Reply:
Right now, it was more of a way to document everything that was happening for me and the SDK team. Things are still up in the air, but I think there's a big chance we'll be moving this to its own repo since we also want this for other languages. We decided to just include this template to the python sdk for ease of testing for the first run.

Reply:
@HipsterBrown We're actually actively working on CLI integration currently so I think it's best to hold off on encouraging use until that's merged. My concern is that the location of this code will likely change (at a minimum the `cookiecutter` code probably will end up in RDK, and the templates could possibly move as well) and encouraging users to interact with it before then could lead to unpleasant user experiences when workflows are suddenly changed from under them. Also, the amount of time where 1) all of this is merged into the Python SDK, but 2) we don't yet have CLI integration, will be on the order of weeks, not months, so I don't think we're holding people back for a significant amount of time.

Let me know if this doesn't sound reasonable to you; happy to chat offline if you want!

Reply:
Thanks for breaking that down. I think having the disclaimer [you suggested](https://github.com/viamrobotics/viam-python-sdk/pull/716#discussion_r1727198707) will be helpful for anyone who happens upon until it is fully ready. 

Given the eventual inclusion in the RDK codebase, I'm curious if a go-based scaffolding library was explored before settling on cookiecutter. But that can be discussed outside of this PR.

Reply:
> Given the eventual inclusion in the RDK codebase, I'm curious if a go-based scaffolding library was explored before settling on cookiecutter.

It was! We looked at a go-based scaffolding library but the difference was `cookiecutter` has tens of thousands of stars and is still maintained, whereas the go library had dozens of stars and (iirc) hadn't been committed to in quite a while. So I think we chose `cookiecutter` because the ecosystem seems more robust and we're less likely to have to maintain it ourselves.